setup/terraform/idp-cluster.tf

```
@@ -98,3 +98,61 @@ resource "humanitec_resource_definition_criteria" "cluster_local" {
     humanitec_resource_definition_criteria.agent
   ]
 }
+
+# Install and configure Humanitec Operator
+
+resource "kubernetes_namespace" "operator_namespace" {
+  metadata {
+    name = var.operator_ns
+  }
+}
+
+resource "tls_private_key" "operator" {
+  algorithm = "RSA"
+  rsa_bits  = 4096
+}
+
+resource "kubernetes_secret" "operator_private_key" {
+  depends_on = [kubernetes_namespace.operator_namespace]
+  metadata {
+    name      = "humanitec-operator-private-key"
+    namespace = var.operator_ns
+  }
+  data = {
+    privateKey              = tls_private_key.operator.private_key_pem
+    humanitecOrganisationID = var.humanitec_org
+  }
+}
+
+resource "humanitec_key" "operator_public_key" {
+  depends_on = [kubernetes_secret.operator_private_key]
+  key = tls_private_key.operator.public_key_pem
+}
+
+resource "kubernetes_namespace" "secret_store_namespace" {
+  metadata {
+    name = "5min-idp-secrets"
+  }
+}
+
+resource "helm_release" "operator" {
+  depends_on = [
+    kubernetes_secret.operator_private_key,
+    kubernetes_namespace.secret_store_namespace,
+    humanitec_key.operator_public_key
+  ]
+  name       = "humanitec-operator"
+  namespace  = var.operator_ns
+  repository = "oci://ghcr.io/humanitec/charts"
+  chart      = "humanitec-operator"
+}
+
+// Hack: registering fake "vault" primary store just to enable Operator Mode ("kubernetes" type is not supported in Orchestrator).
+// Needs to be removed, when Operator Mode becomes the default one.
+resource "humanitec_secretstore" "kubernetes_secret_store" {
+  id      = "5min-idp-secrets"
+  primary = true
```

I think we must set this to `false`. If `true`, then it will change the `primary` store to the 5min-idp store for the duration of my run, or permanently if I don't bother to clean up. In addition, after cleanup the `humanitec` store will become the `primary` on deletion of the 5min-idp store.

Either way, the behavior is quite invasive. I tested it and the flow works fine with the `false` setting and leaves my current `primary` store untouched.
```suggestion
  primary = false
```

---

If we don't set it to `primary`, it will not do the trick: if a user doesn't have any custom primary store they'll fall back to the Legacy Mode.
In your setup it works with `false` because you have already `primary` custom store, in fact you don't need to create this store at all. But I suppose most of the use cases are clean orgs without any secret store. Besides, we can remove it right after we introduce the "legacy" feature flags, this change is only needed not to have a downtime in 5 min IDP while we are rolling out deprecation of the Legacy Mode.

---

I see, we need the store as the fallback trigger for Direct Mode because there are no secret references, right?

It's not perfect, but I agree that the risk is really small of anyone using it with an advanced setup. So let's proceed as you suggest. Thanks for the background!